plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -309,6 +317,140 @@ public void testColumnsNotCachedWhenCacheDisabled()
+
+        // load table for second session
+        assertThat(cachingJdbcClient.getTableStatistics(second, table, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss += 1);
+
+        // read table from cache for second session
+        assertThat(cachingJdbcClient.getTableStatistics(second, table, TupleDomain.all())).isEqualTo(NON_EMPTY_STATS);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit += 1, expectedMiss);
+
+        // cleanup
+        this.jdbcClient.dropTable(SESSION, table);
+    }
+
+    private CachingJdbcClient cachingStatisticsAwareJdbcClient(Duration duration, boolean cacheMissing)
+    {
+        JdbcClient jdbcClient = database.getJdbcClient();
+        JdbcClient statsAwareJdbcClient = new ForwardingJdbcClient() {
+            @Override
+            protected JdbcClient delegate()
+            {
+                return jdbcClient;
+            }
+
+            @Override
+            public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
+            {
+                return NON_EMPTY_STATS;
+            }
+        };
+        return new CachingJdbcClient(statsAwareJdbcClient, SESSION_PROPERTIES_PROVIDERS, duration, cacheMissing);
+    }
+
+    @Test
+    public void testCacheEmptyStatistics()
+    {
+        CachingJdbcClient cachingJdbcClient = createCachingJdbcClient(FOREVER, true);
+        ConnectorSession session = createSession("table");
+
+        JdbcTableHandle table = createTable(new SchemaTableName(schema, "table"));
+
+        int expectedLoad = 0, expectedHit = 0, expectedMiss = 0;
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss);
+
+        // load table
+        assertThat(cachingJdbcClient.getTableStatistics(session, table, TupleDomain.all())).isEqualTo(TableStatistics.empty());
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit, expectedMiss += 1);
+
+        // read from cache
+        assertThat(cachingJdbcClient.getTableStatistics(session, table, TupleDomain.all())).isEqualTo(TableStatistics.empty());
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getStatisticsCacheStats(), expectedLoad, expectedHit += 1, expectedMiss);
+
+        // cleanup
+        this.jdbcClient.dropTable(SESSION, table);
+    }
+
+    @Test
+    public void testGetTableStatisticsDoNotCacheEmptyWhenCachingMissingIsDisabled()
+    {
+        CachingJdbcClient cachingJdbcClient = createCachingJdbcClient(FOREVER, false);
+        ConnectorSession session = createSession("table");
+
+        JdbcTableHandle table = createTable(new SchemaTableName(schema, "table"));
+
+        int expectedLoad = 0, expectedHit = 0, expectedMiss = 0;
```

Comment:
this style is illegal in Trino.

Reply:
@wendigo is removing checkstyle suppressions from this class in https://github.com/trinodb/trino/pull/6808